.github/workflows/main.yml

```diff
@@ -1,9 +1,9 @@
 name: CI Tests
 
-on: [pull_request]
+on: [push, pull_request]
```

Comment:
@ChinouneMehdi Is there a reason you want to run the CI on every push? I guess since Github Actions is "free" it doesn't cost, but I can be overwhelming at times.

Reply:
Sometimes there are commits between the time of pull-request and push events. We should be sure that the commit still pass the tests.
And for someone who push directly to the repos.